test/test.js

```diff
@@ -0,0 +1,245 @@
+		.on('data', function (data) {
+			bFiles.push(data);
+		})
+		.on('end', function () {
+			isDone(done);
+		});
+
+		sass('source/directory with spaces/file with spaces.scss', defaultOptions)
+		.on('data', function (data) {
+			cFiles.push(data);
+		})
+		.on('end', function () {
+			isDone(done);
+		});
+	});
+
+	it('result files aren\'t intermixed when tasks are run simultaniously', function () {
+		assert.equal(aFiles.length, 1);
+		assert.equal(bFiles.length, 1);
+		assert.equal(cFiles.length, 1);
+	});
+});
+
+describe('creates vinyl sourcemaps', function () {
+	this.timeout(20000);
+	var files = [];
+
+	before(function(done) {
+		var options = assign({}, defaultOptions, { sourcemap: true });
+
+		sass('source/file.scss', options)
+		.on('data', function (data) {
+			files.push(data);
+		})
+		.on('end', done);
+	});
+
+	it('does not stream Sass sourcemap files', function () {
+		assert.equal(files.length, 1);
+	});
+
+	it('removes Sass sourcemap comment', function () {
+		assert(
+			files[0].contents.toString().indexOf('sourceMap') === -1,
+			'File contains sourcemap comment'
+		);
+	});
+
+	it('adds vinyl sourcemap', function () {
+		assert.equal(typeof files[0].sourceMap, 'object')
+		assert.equal(files[0].sourceMap.version, 3)
+	});
+
+	it('includes the correct sources', function () {
+		assert.deepEqual(
+			files[0].sourceMap.sources,
+			['_partial.scss', 'file.scss', 'directory/_nested-partial.scss']
+		)
+	});
+
+	describe('for files and directories with spaces', function () {
+		before(function(done) {
+			var options = assign({}, defaultOptions, { sourcemap: true });
+
+			sass('source/directory with spaces/file with spaces.scss', options)
+			.on('data', function (data) {
+				files.push(data);
+			})
+			.on('end', done);
+		});
+
+		it('includes the correct sources', function () {
+			assert.deepEqual(
+				files[1].sourceMap.sources,
+				['file with spaces.scss']
+			);
+		});
+	});
+});
+
+describe('options', function () {
+	this.timeout(20000);
+
+	describe('emitCompileError', function () {
+		var error;
+
+		before(function(done) {
+			var options = assign({}, defaultOptions, { emitCompileError: true });
+
+			sass('source/error.scss', options)
+			.on('data', function () {})
+			.on('error', function (err) {
+				error = err;
+			})
+			.on('end', done);
+		});
+
+		it('emits a gulp error when Sass compilation fails', function () {
+			assert(error instanceof Error);
+			assert.equal(
+				error.message,
+				'Sass compilation failed. See console output for more information.'
+			);
+		});
+	});
+
+	describe('tempDir', function () {
+		var error;
+
+		it('compiles files to a specified directory', function (done) {
+			var source = 'source/file.scss';
+			var tempDir = './custom-temp-dir';
+			var options = assign({}, defaultOptions, { tempDir: tempDir });
+
+			sass(source, options)
+			.on('data', function (data) {
+				var expectedFileLocation = path.join(
```

Comment:
This variable is unused, looks like this test is missing a vinyl file path assertion.